File: 프로그래머스/1/1845. 폰켓몬/폰켓몬.js
function solution(nums) {
    var answer = 0;
    hash_map=new Map()
    
    for(num of nums){
        if ( !(num in hash_map)) { // 없으면은 key값 추가 
            hash_map[num]=true
            
        }
        
    } 
    len_=(Object.keys(hash_map).length)
    
    if (len_ > Object.keys(nums).length /2 ){
        answer=Object.values(nums).length /2
    }
    else
        answer=len_
    
    
    
    
    
    
    
    
    
    
    return answer;
}

File: 프로그래머스/1/1845. 폰켓몬/폰켓몬.java
import java.util.*;
class Solution {
    public int solution(int[] nums) {
        int answer = 0;
        int len=nums.length;
        Set<Integer> set=new HashSet<>();
        
        for(int num : nums){
            set.add(num);
        }
        
        int size=set.size();
        
        if (size >= len/2 ){
            answer=len/2;
        }
        else{
            answer=size;
        }
        
        
        return answer;
    }
}